src/rospkg/os_detect.py

```diff
@@ -126,6 +139,47 @@ def get_codename(self):
             return self.lsb_info[2]
         raise OsNotDetected('called in incorrect OS')
 
+class FdoDetect(OsDetector):
```

Comment:
Fdo seems too specific for this. OsRelease or something similar might make more sense. It looks like it would work on more than just fedora.